crates/pint-deploy/src/main.rs

```diff
@@ -0,0 +1,73 @@
+use clap::Parser;
+use essential_rest_client::builder_client::EssentialBuilderClient;
+use essential_types::contract::Contract;
+use std::path::PathBuf;
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+/// Tool to deploy contracts in a Pint project to a Essential builder endpoint.
+struct Args {
+    /// The endpoint of builder to bind to.
+    #[arg(long)]
+    builder_address: String,
+    /// The path to the package manifest.
+    ///
+    /// If not provided, the current directory is checked and then each parent
+    /// recursively until a manifest is found.
+    #[arg(long)]
+    manifest_path: Option<PathBuf>,
+}
+
+#[tokio::main]
+async fn main() {
+    let args = Args::parse();
+    if let Err(err) = run(args).await {
+        eprintln!("Command failed because: {}", err);
+    }
+}
+
+async fn run(args: Args) -> anyhow::Result<()> {
+    let Args {
+        builder_address,
+        manifest_path,
+    } = args;
+    let builder_client = EssentialBuilderClient::new(builder_address)?;
+    let contract_path = find_manifest(manifest_path)?;
```

Comment:
Just adding a note for future reference:

In the future, we can use this logic along with the `pint build` args to automatically

1. build a project and
2. infer where the output `<contract>.json` file should be

However in the meantime, we can just require a `--contract <path-to-contract.json>` arg and save the build logic for follow-up.